pkg/network/ovn_kubernetes.go

```diff
@@ -521,25 +527,23 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 		fillKubeProxyDefaults(conf, nil)
 	}
 
-	zoneModeMigrationIsOngoing := isZoneModeMigrationAboutToStartOrOngoing(bootstrapResult.OVN, &targetZoneMode)
```

Comment:
what is the difference between `zoneModeMigrationIsOngoing` and `ongoingMigrationFromSingleToMultiZone`

Reply:
AH I see `zoneModeMigrationIsOngoing` had checks to go from multizone to single zone so we don't need this anymore but still end to end what's the current diff here ?

Reply:
yes, the code included checks for the migration in the two directions

Reply:
As for the current diff, CNO considers it's moving from single to multizone according to the existing conditions on OVNK status + the presence of the configmap. Since the configmap is removed by CNO as soon as the upgrade to IC completes, CNO won't mistakenly consider that it has to redo the single->multizone "migration" when transient conditions on any ovnkube-node instance make it look like an upgrade phase happening again.

So we're picking YAMLs from the single zone folder if the configmap says the target is single zone; we're picking YAMLs from the multizone-tmp folder if the conditions for the single->multizone phase hinted at in the  previous paragraph apply (ovnk status + config map); if none of that applies, we're picking YAMLs from the multizone folder, which is our final state. 

When we're not picking yamls from the multizone folder, `ongoingUpgradeToInterconnect` is true.

Reply:
ack thanks for explaining this to me.